app/org/sagebionetworks/bridge/validators/ScheduleValidator.java

```diff
@@ -16,86 +21,133 @@
+        if (oneTimeScheduleHasInterval(schedule)) {
+            errors.reject("executing once should not have an interval");
+        }
+        if (recurringScheduleLacksRepeatingInfo(schedule)) {
+            errors.reject("recurring schedules should have either a cron expression, or an interval");
+        }
+        if (cronExpressionInvalid(schedule)) {
+            errors.rejectValue(Schedule.CRON_TRIGGER_PROPERTY, "is an invalid cron expression");
+        }
+        if (intervalMissingTimes(schedule)) {
+            errors.rejectValue(Schedule.TIMES_PROPERTY, "are required for interval-based schedules");
+        }
+        if (intervalTooShort(schedule.getInterval())) {
+            errors.rejectValue(Schedule.INTERVAL_PROPERTY, "must be at least one day");
+        }
+        if (periodTooShort(schedule.getExpires())) {
+            errors.rejectValue(Schedule.EXPIRES_PROPERTY, "must be at least one hour");
+        }
+        if (delayWithTimesAreAmbiguous(schedule.getDelay(), schedule.getTimes())) {
+            errors.rejectValue(Schedule.DELAY_PROPERTY, "is less than one day, and times of day are also set for this schedule, which is ambiguous");
+        }
+        if (timesNotOrderedChronologically(schedule)) {
+            errors.rejectValue(Schedule.ENDS_ON_PROPERTY, "should be at least an hour after the startsOn time");
+        }
+        validateActivities(schedule, errors);
+    }
+
+    private boolean oneTimeScheduleHasInterval(Schedule schedule) {
+        return (schedule.getScheduleType() == ScheduleType.ONCE && schedule.getInterval() != null);
+    }
+    
+    private boolean recurringScheduleLacksRepeatingInfo(Schedule schedule) {
+        boolean bothOrNeither = (schedule.getInterval() != null && isNotBlank(schedule.getCronTrigger()) || 
+                                (schedule.getInterval() == null && isBlank(schedule.getCronTrigger())) );
+        return (schedule.getScheduleType() == ScheduleType.RECURRING) && bothOrNeither;
+    }        
+
+    private boolean cronExpressionInvalid(Schedule schedule) {
+        return schedule.getCronTrigger() != null && !CronExpression.isValidExpression(schedule.getCronTrigger());
+    }
+    
+    private boolean intervalMissingTimes(Schedule schedule) {
+        return schedule.getInterval() != null && (schedule.getTimes() == null || schedule.getTimes().isEmpty());
+    }
+    
+    /**
+     * Interval currently must be at least 24 hours (this is because it is ambiguous to have an interval under
+     * twenty-four hours along with specific times of the day). For intervals under a day, a cron expression may work.
+     * 
+     * @param interval
+     * @return
+     */
+    private boolean intervalTooShort(Period interval) {
+        if (interval == null) {
+            return false;
+        }
+        DateTime now = DateTime.now();
+        DateTime dur = now.plus(interval);
+        DateTime hours24 = now.plusHours(24);
+        return dur.isBefore(hours24);
```

Comment:
This will fail when Daylight Savings starts (the one day of the year that's only 23 hours long). The safest thing to do would be to check that the period has at least 1 in getYears() or getMonths() or getWeeks() or getDays() (and optionally has none of getHours(), getMinutes(), getSeconds(), and getsMillis(), unless we decide that a period of 1 day and 30 minutes makes sense.)